src/Mvc/Mvc.TagHelpers/src/PrerenderingHelpers.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    internal static class PrerenderingHelpers
```

Comment:
```suggestion
    internal static class ComponentsPrerenderingSectionsStore
```
or something like that `*Helpers` are usually bad names because they don't convey what's it about and it's easy for them to turn in to a dumping ground of random features.

Reply:
I've called it `ComponentDeferredContentStore` because with the new changes, it's not necessarily specific to prerendering. Does that sound ok?